osu.Game/Screens/Play/HUD/GameplayLeaderboard.cs

```diff
@@ -46,42 +75,114 @@ protected override void LoadComplete()
+        public ILeaderboardScore Add([CanBeNull] User user, bool isTracked)
         {
             var drawable = CreateLeaderboardScoreDrawable(user, isTracked);
 
+            if (isTracked)
+            {
+                if (trackedScore != null)
+                    throw new InvalidOperationException("Cannot track more than one scores.");
+
+                trackedScore = drawable;
+            }
+
             drawable.Expanded.BindTo(Expanded);
 
-            base.Add(drawable);
+            Flow.Add(drawable);
             drawable.TotalScore.BindValueChanged(_ => sorting.Invalidate(), true);
 
-            Height = Count * (GameplayLeaderboardScore.PANEL_HEIGHT + Spacing.Y);
+            int displayCount = Math.Min(Flow.Count, maxPanels);
+            Height = displayCount * (GameplayLeaderboardScore.PANEL_HEIGHT + Flow.Spacing.Y);
+            requiresScroll = displayCount != Flow.Count;
 
             return drawable;
         }
 
+        public void Clear()
+        {
+            Flow.Clear();
+            trackedScore = null;
+            scroll.ScrollToStart(false);
+        }
+
         protected virtual GameplayLeaderboardScore CreateLeaderboardScoreDrawable(User user, bool isTracked) =>
             new GameplayLeaderboardScore(user, isTracked);
 
-        public sealed override void Add(GameplayLeaderboardScore drawable)
+        protected override void Update()
         {
-            throw new NotSupportedException($"Use {nameof(AddPlayer)} instead.");
+            base.Update();
+
+            if (requiresScroll && trackedScore != null)
+            {
+                float scrollTarget = scroll.GetChildPosInContent(trackedScore) + trackedScore.DrawHeight / 2 - scroll.DrawHeight / 2;
+                scroll.ScrollTo(scrollTarget, false);
+            }
+
+            const float panel_height = GameplayLeaderboardScore.PANEL_HEIGHT;
+
+            float fadeBottom = scroll.Current + scroll.DrawHeight;
+            float fadeTop = scroll.Current + panel_height;
+
+            if (scroll.Current <= 0) fadeTop -= panel_height;
+            if (!scroll.IsScrolledToEnd()) fadeBottom -= panel_height;
+
+            // logic is mostly shared with Leaderboard, copied here for simplicity.
+            foreach (var c in Flow.Children)
+            {
+                float topY = c.ToSpaceOfOtherDrawable(Vector2.Zero, Flow).Y;
+                float bottomY = topY + panel_height;
+
+                bool requireTopFade = requiresScroll && topY <= fadeTop;
+                bool requireBottomFade = requiresScroll && bottomY >= fadeBottom;
+
+                if (!requireTopFade && !requireBottomFade)
+                    c.Colour = Color4.White;
+                else if (topY > fadeBottom + panel_height || bottomY < fadeTop - panel_height)
+                    c.Colour = Color4.Transparent;
+                else
+                {
+                    if (bottomY - fadeBottom > 0 && requiresScroll)
```

Comment:
This condition is pretty much `requireBottomFade`, isn't it? (Well, maybe save for the fact that the inequality is strict.) Might as well reuse that?

Reply:
This was copied across from `Leaderboard` quite verbatim, but I'll fix it here sure.